### ISSUE TITLE
where is box_downsample_halide.h . file

### ISSUE THREAD

**Issue by soham24** [PK15946/Halide-Android-example#1]
Thanks for this example.
but I am having trouble at compiling it.

I see box_downsample_halide.h . file in included in the source but not in this repo.
where  I can get it?

**Comment by soham24** [comment]
@PK15946

**Comment by PK15946** [comment]
@soham24 Yes, this repo didn't include that head file. Some head files would be generated by the halide generator and box_downsample_halide.h is one of them. But it should appear when the cmake process is finished.

**Comment by soham24** [comment]
@PK15946 thank you for the reply.
I have built halide with  make file command 👍   % make -f ../Halide/Makefile
but It did not generate any box_downsample_halide.h file.
am I missing something here?

**Comment by PK15946** [comment]
Sorry I didn't make it clearly. "box_downsample_generator.cpp" is called a halide generator, after it was compiled and executed, there will appear two files, one of them is box_downsample_halide.h. And when you build the android project,  box_downsample_generator.cpp should be automatically compiled and executed if everything goes well.

**Comment by soham24** [comment]
Thanks. 
I am facing this weird error now.
  In file included from ...../hl-pipelines/box_downsample_generator.cpp:14:
 /halide_build/include/Halide.h:17:10: fatal error: 'string' file not found
  #include <string>
           ^

**Comment by PK15946** [comment]
It looks like an error about your path of c++ head files, I can't help you much about this.

**Comment by soham24** [comment]
@PK15946   I will compile halide freshly, and give it a try.
thank you so much for your help.

**Comment by soham24** [comment]
I see 2 cmake files
1 in app folder and other in hl-pipelines folder. 
The previous issue was because I was directly running apps cmake file. so remaining files did not generate.


Can you tell me order to run both cmake files.
thanks.

**Comment by PK15946** [comment]
@soham24 Just download a android studio, import the provided project and sync it, done!

**Comment by soham24** [comment]
Ok. One question. Does I need to set special Android version for buid?
For build, tools and platform tools etc?
As I have seen halide has recommended Android 21 in their official example
